app/src/androidTest/java/de/test/antennapod/ui/PlaybackTest.java

```diff
@@ -269,4 +302,66 @@ public void testReplayEpisodeContinuousPlaybackOn() throws Exception {
     public void testReplayEpisodeContinuousPlaybackOff() throws Exception {
         replayEpisodeCheck(false);
     }
+
+    @Test
+    public void testSmartMarkAsPlayed_Skip_Average() throws Exception {
+        doTestSmartMarkAsPlayed_Skip_ForEpisode(0);
+    }
+
+    @Test
+    public void testSmartMarkAsPlayed_Skip_LastEpisodeInQueue() throws Exception {
+        doTestSmartMarkAsPlayed_Skip_ForEpisode(-1);
+    }
+
+    private void doTestSmartMarkAsPlayed_Skip_ForEpisode(int itemIdxNegAllowed) throws Exception {
+        setSmartMarkAsPlayedPreference(60);
+        // ensure when an episode is skipped, it is removed due to smart as played
+        setSkipKeepsEpisodePreference(false);
+
+        uiTestUtils.addLocalFeedData(true);
+
+        int fiIdx;
+        if (itemIdxNegAllowed >= 0) {
+            fiIdx = itemIdxNegAllowed;
+        } else { // negative index: count from the end, with -1 being the last one, etc.
+            fiIdx = DBReader.getQueue().size() + itemIdxNegAllowed;
+        }
+        final FeedItem feedItem = DBReader.getQueue().get(fiIdx);
+
+        gotoQueueScreen();
+        playFromQueue(fiIdx);
+
+        skipEpisode();
+        Thread.sleep(1000); // ensure the skip is processed
```

Comment:
I don't like this, to be honest. Tests already take ages to complete.

Reply:
I'll look into it. The test code needs to get a `PlaybackService` instance and query its state.

Reply:
The fix with Awaitility makes one of the primary `assert`s buried within Awaitility calls. 

It's slightly less readable but I think it's acceptable.

```java
        //  assert item no longer in queue (needs to wait till skip is asynchronously processed)
        Awaitility.await()
                .atMost(1000, MILLISECONDS)
                .untilAsserted(() -> {
                    assertThat("Ensure smart mark as play will lead to the item removed from the queue",
                            DBReader.getQueue(), not(hasItems(feedItem)));
                });
```